app/extras/queries/visible_discussions.rb

```diff
@@ -86,10 +94,9 @@ def sorted_by_latest_activity
     self
   end
 
-  def sorted_by_latest_motions
-    @relation = @relation.joined_to_current_motion
-                         .preload(:current_motion, {group: :parent})
-                         .order('motions.closing_at ASC, last_activity_at DESC')
+  def sorted_by_importance
+    join_to_starred_motions && join_to_motions
+    @relation = @relation.order('smo.closing_at ASC, mo.closing_at ASC, dv.starred DESC NULLS LAST, last_activity_at DESC')
```

Comment:
It's null when you're left joining on a table and it can't find a left join
though.

On Mon, Aug 3, 2015 at 2:09 AM, Robert Guthrie <notifications@github.com>
wrote:

> In app/extras/queries/visible_discussions.rb
> <https://github.com/loomio/loomio/pull/2369#discussion_r36067498>:
>
> > @@ -86,10 +94,9 @@ def sorted_by_latest_activity
> >      self
> >    end
> >
> > -  def sorted_by_latest_motions
> > -    @relation = @relation.joined_to_current_motion
> > -                         .preload(:current_motion, {group: :parent})
> > -                         .order('motions.closing_at ASC, last_activity_at DESC')
> > +  def sorted_by_importance
> > +    join_to_starred_motions && join_to_motions
> > +    @relation = @relation.order('smo.closing_at ASC, mo.closing_at ASC, dv.starred DESC NULLS LAST, last_activity_at DESC')
>
> fyi - starred cannot be null
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2369/files#r36067498>.
>